ctf/core.py

```diff
@@ -29,8 +29,12 @@ def ensure_active():
 
 
 def get_teams():
-    # TODO: Team.points ought to be a SQL sum instead of denormalized
-    return Team.query.order_by(Team.points.desc(), Team.last_fleg).all()
+    # This is pure mystical alchemy
+    q = (db.session.query(Team, db.func.ifnull(db.func.sum(Challenge.points),
+                                               0).label('points'))
+         .outerjoin(Solve).outerjoin(Challenge).group_by(Team.id)
+         .order_by(db.desc('points'), db.func.min(Solve.earned_on), Team.id))
+    return q.all()
```

Comment:
I started to think that, but we have get functions in core that are similar to this. Might be a larger design decision we should make